pyttb/ktensor.py

```diff
@@ -49,25 +60,24 @@ class ktensor(object):
 
     __slots__ = ("weights", "factor_matrices")
 
-    def __init__(self, factor_matrices=None, weights=None, copy=True):
+    def __init__(
+        self,
+        factor_matrices: Optional[List[np.ndarray]] = None,
+        weights: Optional[np.ndarray] = None,
+        copy: bool = True,
```

Comment:
Should all optional parameters use `Optional[...]`? Why is `copy` not typed as `Optional[bool]`?

Reply:
`Optional[SomeType]` is equivalent to `Union[NoneType, SomeType]`. For `copy` we only process bools but provide a default value. Slightly different interpretations of optional.

Reply:
Got it! That makes sense.